RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -207,15 +198,25 @@ void PileupJetIdAlgo::runMva()
                           if(std::abs(internalId_.jetEta_) > jEtaMax_.at(nEtaBins_-1)) {
                               internalId_.mva_ = -2.;
                           } else {
-                            for(int v=0; v<nEtaBins_;v++){
-                                if(std::abs(internalId_.jetEta_)>=jEtaMin_.at(v) && std::abs(internalId_.jetEta_)<jEtaMax_.at(v)){
-                                    internalId_.mva_ = etaReader_.at(v)->EvaluateMVA( tmvaMethod_.c_str() );
+                            for(int v=0; v<nEtaBins_; v++){
+                                if(std::abs(internalId_.jetEta_)>=jEtaMin_.at(v) && std::abs(internalId_.jetEta_)<jEtaMax_.at(v)) {
+                                    std::vector<float> vars;
+                                    for(std::vector<std::string>::iterator it=tmvaEtaVariables_.at(v).begin(); it!=tmvaEtaVariables_.at(v).end(); ++it) {
+                                        std::pair<float *,float> var = variables_.at((*it).c_str());
+                                        vars.push_back( *var.first );
+                                    }
+                                    internalId_.mva_ = etaReader_.at(v)->GetClassifier(vars.data());
                                     break;
                                 }
                             } 
                           }
 			} else {
-			  internalId_.mva_ = reader_->EvaluateMVA( tmvaMethod_.c_str() );
+                            std::vector<float> vars;
+                            for(std::vector<std::string>::iterator it=tmvaVariables_.begin(); it!=tmvaVariables_.end(); ++it) {
+                                std::pair<float *,float> var = variables_.at((*it).c_str());
+                                vars.push_back( *var.first ); 
+                            }       
+                            internalId_.mva_ = reader_->GetClassifier(vars.data());
```

Comment:
Again there is duplicated code here. As in the my comment above, you should create a new function that can be called by each part of the `if-else`.